src/sentry/api/endpoints/organization_metrics.py

```diff
@@ -137,12 +135,7 @@ def get(self, request: Request, organization, tag_name) -> Response:
                 use_case_id=get_use_case_id(request),
             )
         except (InvalidParams, DerivedMetricParseException) as exc:
-            msg = str(exc)
-            # TODO: Use separate error type once we have real data
-            if "Unknown tag" in msg:
-                raise ResourceDoesNotExist(f"tag '{tag_name}'")
```

Comment:
I really dislike this approach, not sure you can do anything about it but
if it is a custom error (i.e. we can modify it without too much trouble) , I would put a member dedicated to this ( so I don't need to look through a string based on the representation of the error), this would make this handling
more robust and easier to maintain.

Reply:
This wasn't something I implemented, I have removed it since the endpoint is unused. Will do better exception handling in case we decide to stick to this endpoint, since in the future we might move to a completely new implementation.